### DISCUSSION TITLE
module allsky_dewheater expansion with the “sensor” to include Solo Cloudwachter

### DISCUSSION THREAD

**Discussion by Adler6907** [AllskyTeam/allsky-modules#38] [Show and tell]
I have expanded the `allsky_dewheater` module to include a “sensor”. The new "sensor" reads the weather data from the weather station "Solo Cloudwatcher" from the company "Lunatico Astronomia" (lunaticoastro.com) via HTTP. 

Runs on an Rpi5 with Bookwoorm.
new libraries `import gpiod`,  `import urllib.request` and `import json` are needed (they were already installed on mine)
new function `def readSolo(url)`, Changes in `metaData`, `getSensorReading` and `dewheater`

![image](https://github.com/AllskyTeam/allsky-modules/assets/119796992/6b25357c-15b6-4ee1-ac63-a1b14ecdd74c)

Here is the data provided by the “Solo Cloudwatcher”:
`{ "LastReadings": {
"dataGMTTime" : "2023/12/18 18:22:07",
"cwinfo" : "Serial: 2550, FW: 5.89",
"clouds" : -18.130000,
"cloudsSafe" : "Safe",
"temp" : 7.820000,
"wind" : 7,
"windSafe" : "Safe",
"gust" : 8,
"rain" : 3100,
"rainSafe" : "Safe",
"lightmpsas" : 19.82,
"lightSafe" : "Safe",
"switch" : 1,
"safe" : 1,
"hum" : 40,
"humSafe" : "Safe",
"dewp" : -4.940000,
"rawir" : -22.680000,
"abspress" : 998.450000,
"relpress" : 1027.527780,
"pressureSafe" : "Safe"
}
}`

Here in the zip file is also another module "allsky_solo_cloudwatcher" which writes all the data listed above from the "Solo-Cloudwatcher" to `/home/pi/allsky/tmp/allskydb.py` for possible use in other modules

[allsky_dewheater.zip](https://github.com/AllskyTeam/allsky-modules/files/13728925/allsky_dewheater.zip)

CS

**Comment by Alex-developer** [comment]
Thats really nice :-)

We are currently freezing development until we release the Point-Release-4 which is planned for this Fri/Sat. As soon as that is released I will get this merged into the code

Thanks for the contribution :-)

Alex

**Comment by Adler6907** [comment]
ohha, in the script "allsky_dewheater.py" in the ZIP file above, there is a small error in the call to "readSolo" in the function "getSensorReading"

instead of
```
    elif sensorType == "SOLO-Cloudwatcher":
        temperature, humidity, pressure, dewPoint, = readSolo(url)  
```

it has to look like this (a comma after "dewPoint" is too many)
```
    elif sensorType == "SOLO-Cloudwatcher":
        temperature, humidity, pressure, dewPoint = readSolo(url)  
```

I changed it in the ZIP file above

**Comment by Alex-developer** [comment]
Closing this as dealing with it as an issue

See https://github.com/AllskyTeam/allsky-modules/issues/39

**Comment by Adler6907** [comment]
Suggestion: You could add a “sensor” that, like the Solo, gets the weather data from a surrounding weather station with an API key from weather.com. Here is a [documentary ](https://docs.google.com/document/d/1KGb8bTVYRsNgljnNH67AMhckY8AQT2FVwZ9urj8SWBs/edit) about it. This is how I get the wind direction because the SOLO doesn't have that (yet).

I could add this to the SOLO code in the next few days if desired.

This means that users of the Allsky software can use environmental data without physically installing a sensor
